cmd/operator-sdk/bundle/validate_test.go

```diff
@@ -0,0 +1,131 @@
+
+  # Generate manifests locally.
+  $ operator-sdk bundle create --generate-only
+
+  # Validate the directory containing manifests and metadata.
+  $ operator-sdk bundle validate ./deploy/olm-catalog/test-operator
+
+To build and validate an image:
+
+  # Create a registry namespace or use an existing one.
+  $ export NAMESPACE=<your registry namespace>
+
+  # Build and push the image using the docker CLI.
+  $ operator-sdk bundle create quay.io/$NAMESPACE/test-operator:v0.1.0
+  $ docker push quay.io/$NAMESPACE/test-operator:v0.1.0
+
+  # Ensure the image with modified metadata and Dockerfile is valid.
+  $ operator-sdk bundle validate quay.io/$NAMESPACE/test-operator:v0.1.0
+
+`))
+
+			flag := cmd.Flags().Lookup("image-builder")
+			Expect(flag).NotTo(BeNil())
+			Expect(flag.Shorthand).To(Equal("b"))
+			Expect(flag.DefValue).To(Equal("docker"))
+			Expect(flag.Usage).To(Equal("Tool to extract bundle image data. " +
+				"Only used when validating a bundle image. One of: [docker, podman]"))
+
+			flag = cmd.Flags().Lookup("output")
+			Expect(flag).NotTo(BeNil())
+			Expect(flag.Shorthand).To(Equal("o"))
+			Expect(flag.DefValue).To(Equal(internal.Text))
+			Expect(flag.Usage).To(Equal("Result format for results. One of: [text, json-alpha1]"))
+		})
+	})
+
+	Describe("Validate", func() {
+		var cmd bundle.ValidateCmd
+		BeforeEach(func() {
+			cmd = bundle.ValidateCmd{}
+		})
+
+		It("fails with no args", func() {
+			err := cmd.Validate([]string{})
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).To(Equal("an image tag or directory is a required argument"))
+		})
+		It("fails with more than one arg", func() {
+			err := cmd.Validate([]string{"a", "b"})
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).To(Equal("an image tag or directory is a required argument"))
+		})
+
+		It("fails if the output format isnt text or json-alpha1", func() {
+			wrongArg := "json-alpha2"
+			cmd.OutputFormat = wrongArg
+			err := cmd.Validate([]string{"quay.io/person/example"})
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).To(Equal("invalid value for output flag: " + wrongArg))
+		})
+
+		It("fails if the output format isn't docker or podman", func() {
+			wrongArg := "foobar"
+			cmd.ImageBuilder = wrongArg
+			cmd.OutputFormat = "text"
+			err := cmd.Validate([]string{"quay.io/person/example"})
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).To(Equal("unrecognized image-builder option: " + wrongArg))
+
+		})
+
+		It("succeeds if the arg is text or json-alpha1 and output format is docker or podman", func() {
+			cmd.OutputFormat = "text"
+			cmd.ImageBuilder = "docker"
+			err := cmd.Validate([]string{"quay.io/person/example"})
+			Expect(err).NotTo(HaveOccurred())
+
+			cmd.OutputFormat = "json-alpha1"
+			cmd.ImageBuilder = "podman"
+			err = cmd.Validate([]string{"quay.io/person/example"})
+			Expect(err).NotTo(HaveOccurred())
+		})
+	})
+	PDescribe("Run", func() {
+	})
```

Comment:
why it is required?

Reply:
I put in pending tests just as a reminder that they need to be implemented, but was going to wait until after kubebuilder is merged in because it'll require some additional refactoring. I can remove them if you think that's not needed.

Reply:
could you please add // todo(jberkhahn):  {desc of what should be done} or remove them ?